homeassistant/components/opnsense/__init__.py

```diff
@@ -36,42 +31,43 @@
 )
 
 
-def setup(hass, config):
-    """Set up the opnsense component."""
+@dataclass
+class OPNsenseData:
+    """Shared OPNsense data."""
 
-    conf = config[DOMAIN]
-    url = conf[CONF_URL]
-    api_key = conf[CONF_API_KEY]
-    api_secret = conf[CONF_API_SECRET]
-    verify_ssl = conf[CONF_VERIFY_SSL]
-    tracker_interfaces = conf[CONF_TRACKER_INTERFACE]
+    hass_config: dict
 
-    interfaces_client = diagnostics.InterfaceClient(
-        api_key, api_secret, url, verify_ssl
-    )
-    try:
-        interfaces_client.get_arp()
-    except APIException:
-        _LOGGER.exception("Failure while connecting to OPNsense API endpoint")
-        return False
 
-    if tracker_interfaces:
-        # Verify that specified tracker interfaces are valid
-        netinsight_client = diagnostics.NetworkInsightClient(
-            api_key, api_secret, url, verify_ssl
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the OPNsense component."""
+    if config.get(DOMAIN) is not None:
+        OPNsenseData.hass_config = config
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": SOURCE_IMPORT}, data=config[DOMAIN]
+            )
         )
-        interfaces = list(netinsight_client.get_interfaces().values())
-        for interface in tracker_interfaces:
-            if interface not in interfaces:
-                _LOGGER.error(
-                    "Specified OPNsense tracker interface %s is not found", interface
-                )
-                return False
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up the OPNsense from a config entry."""
+
+    api_data = {
+        CONF_API_KEY: entry.data[CONF_API_KEY],
+        CONF_API_SECRET: entry.data[CONF_API_SECRET],
+        "base_url": entry.data[CONF_URL],
+        "verify_cert": entry.data[CONF_VERIFY_SSL],
+    }
+    tracker_interfaces = entry.data.get(CONF_TRACKER_INTERFACE)
 
     hass.data[OPNSENSE_DATA] = {
```

Comment:
```suggestion
    hass.data[OPNSENSE_DATA][entry.entry_id] = {
```

We usually store these by entry id since the user may have more than one of the integration.